api/node_pool.go

```diff
@@ -47,3 +50,16 @@ func (np NodePool) Taints() []*corev1.Taint {
 	}
 	return taints
 }
+
+func (np NodePool) KarpenterRequirements() []v1.NodeSelectorRequirementApplyConfiguration {
```

Comment:
Where are these requirements actually going to be used? Will that logic come later?

Reply:
there is an example of how to define the requirements in the PR description

this is how to be used
https://github.com/zalando-incubator/kubernetes-on-aws/pull/7617/files#diff-74af8dd6cb1e9f4e450b66a4de808737ccef4387b16b6a3a81824e973b5f2e2dR165

Reply:
Got it. Thanks!